core/src/main/scripts/migrate_db.py

```diff
@@ -209,6 +210,42 @@ def check_reference_genome(portal_properties, cursor, force_migration):
         if not force_migration:
             sys.exit(1)
 
+def check_and_exit_if_fusions(cursor):
+    try:
+        # look for any fusion records
+        # TODO is this OK or maybe we could have fusions from mutation files?
+        cursor.execute(
+            """
+                SELECT COUNT(*)
+                FROM mutation_event
+                WHERE MUTATION_TYPE = "Fusion";
+            """)
+        fusion_count = cursor.fetchone()
+        if (fusion_count[0] >= 1):
+            print('Found %i fusions in the mutation_event table.  Fusions are being replaced by structural variants. Any study with fusions need to be dropped.' % (fusion_count), file=ERROR_FILE)
+            # get the list of studies that need to be cleaned up
+            cursor.execute(
+                """
+                    SELECT cancer_study.CANCER_STUDY_IDENTIFIER, COUNT(mutation.MUTATION_EVENT_ID)
+                    FROM cancer_study,
+                        genetic_profile
+                    LEFT JOIN mutation ON genetic_profile.GENETIC_PROFILE_ID = mutation.GENETIC_PROFILE_ID
+                    LEFT JOIN mutation_event ON mutation.MUTATION_EVENT_ID = mutation_event.MUTATION_EVENT_ID
+                    WHERE 
+                        genetic_profile.CANCER_STUDY_ID = cancer_study.CANCER_STUDY_ID
+                        AND mutation_event.MUTATION_TYPE = "Fusion"
+                    GROUP BY cancer_study.CANCER_STUDY_IDENTIFIER
+                    HAVING count(mutation.MUTATION_EVENT_ID) > 0
+                """)
+            rows = cursor.fetchall()
+            print("The following studies have fusions in the mutation_event table:", file=ERROR_FILE)
+            for row in rows:
+                print("\t%s" % (row[0]), file=ERROR_FILE)
+            sys.exit(1)
+    
+    except MySQLdb.Error as msg:
+        print(msg, file=ERROR_FILE)
```

Comment:
maybe in functions like this .. (where a necessary condition is checked) .. we should exit the migration if there is a mysql error. We would not want the modification to proceed if the check cannot be run. (see other functions below)